### PR TITLE
Disp team index cleanup

### PR DESCRIPTION
**Pull request by Vultraz**
Removes the hideously confusing distinction between `viewing/playing_team` (the vector index) and `viewing/playing_side` (the side number). We only expose the index and query the team object for the side number, if needed.

### CODE REVIEW

**Comment by CelticMinstrel** [comment]
This is basically the same concept as #8271, right?

**Comment by Vultraz** [comment]
Yes, but less intrusive and removes a lot of instances of manual vector access.

File: src/play_controller.cpp [hunk 1]
```diff
@@ -378,8 +378,11 @@ void play_controller::fire_start()
 	// start event may modify start turn with WML, reflect any changes.
 	gamestate().gamedata_.get_variable("turn_number") = static_cast<int>(turn());
 
-	refresh_objectives();
-	check_objectives();
+	// We might have no teams here, and if so, we don't want to crash
+	if(!get_teams().empty()) {
```
Comment (Vultraz) [inline src/play_controller.cpp:382]: @gfgtdf would this be the right fix? With my changes, `display::viewing_side()` attempts vector access, and we don't want to crash at this point if no teams are defined.

Comment (Vultraz) [reply src/play_controller.cpp]: Nevermind, I just added a check to refresh_objectives since check_objectives already has one.

**Comment by Vultraz** [comment]
All tidied up! I'd like to get this merged before I do further refactoring.

**Comment by gfgtdf** [comment]
I'm kinda on vacation right now with no access to my PC so I won't able able to do a full review soon.

About this first commit: I'd think the main reason team_valid exists is that the team vector might be empty, not the the team index is too high.

**Comment by Vultraz** [comment]
Hmm... That seems plausible. That shouldn't be a case in normal circumstances, though, so I think I'll just leave that out. Or I could use `.at()`, but it seems better to let the error case error.

**Comment by CelticMinstrel** [comment]
Sounds like not removing team_valid would've been helpful for your crash case that you fixed later on…

**Comment by Vultraz** [comment]
Probably not, since the crash was in a place that wasn't checked before since before there was no vector access at all.

**Comment by CelticMinstrel** [comment]
My point is that the function _could've_ been used there to check it.

**Comment by Vultraz** [comment]
I shall consider re-adding it.

**Comment by Vultraz** [comment]
Merging this now since celmin said he has no objections.

**Pull request merged**